### DISCUSSION TITLE
Why the latter class not override the previous one?

### DISCUSSION THREAD

**Discussion by leslie555** [tailwindlabs/tailwindcss#10856] [Help] [answered]
It always use the class `ml-8` in the first child `div`.
https://play.tailwindcss.com/LtQxOz4co5

Update:
I guess the rule is to compare the first number after `-`(dash).

**Comment by wongjn** [comment, marked as answer]
Generally, you should avoid applying multiple classes of the same property with the same variant, i.e. `ml-8 ml-0` is not advised while `ml-8 first:ml-0` would be fine.

Your issue arises from the fact that the CSS rule that is defined the latest in the stylesheet wins out when the rules have the same specificity.

**Reply by leslie555** [reply]
I viewed the generated styles, the order of these classes are sorted by `sort()`.

**Reply by adamwathan** [reply]
We only sort them to make it easier to maintain our test suite. Definitely shouldn’t apply two conflicting classes to the same element.
